### PR TITLE
Restore NONE as an enum value for typedef community-type

### PR DESCRIPTION
**Pull request by dplore**
Proposed fix for https://github.com/openconfig/public/issues/945

### Change Scope
In the current OC models, there is ambiguity regarding how to use send-community-type to disable sending any communities at the bgp global, peer-group and neighbor levels.  The OC model description says the lack of the `send-community-type` node means communities should not be sent.  This is in conflict with the expected inheritance from global, peer-group and neighbor levels in BGP configuration.   With inheritance, the lack of a node means to inherit the configuration from the next higher level.

It was proposed in #945 that an explicitly defined send-community-type as an empty list could be used to indicate no communities should be sent.  But this is problematic for some encodings and likely toolchains because [RFC7590](https://datatracker.ietf.org/doc/html/rfc7950) states that XML encoded yang simply omits empty lists ([ref](https://github.com/openconfig/public/issues/945#issuecomment-2311652367)).

This proposal removes the deprecated status from the `NONE` value of the `community-type` enum.

### CODE REVIEW

File: release/models/bgp/openconfig-bgp-types.yang [hunk 1]
```diff
@@ -795,10 +801,8 @@ module openconfig-bgp-types {
       }
       enum NONE {
         description
-            "Do not send any community attribute.
-            This value has been deprecated because the node is now
-            a leaf-list.";
-        status deprecated;
+            "Do not send any community attribute. If this value is present
+            then no communities will be sent.";
```
Comment (rszarecki) [inline release/models/bgp/openconfig-bgp-types.yang:805]: This value must not coexist with any other value in same community type list.
[NONE STANDARD] is ambiguous.

**Review by rszarecki — CHANGES_REQUESTED** [review]

File: release/models/bgp/openconfig-bgp-types.yang (hunk 1 above)
Comment (dplore) [reply release/models/bgp/openconfig-bgp-types.yang]: Done

**Review by s19nal — APPROVED** [review]
as discussed at OC operators meeting - removing a deprecated status from one enum in the leaf looks to be ok although it might be slightly contentious to call it non-breaking.
However re-instating the deprecated NONE enum solves the problem that was introduced by removing it in the first place, so I'm happy :)

**Comment by nokia1adam** [comment]
@dplore, thanks for resolving this issue.

**Pull request merged**